libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -38,6 +38,41 @@ ContractDefinition const& IRGenerationContext::mostDerivedContract() const
 	return *m_mostDerivedContract;
 }
 
+ContractDefinition const* IRGenerationContext::superContract(ContractDefinition const& _contract) const
+{
+	auto const& hierarchy = mostDerivedContract().annotation().linearizedBaseContracts;
+	auto it = find(hierarchy.begin(), hierarchy.end(), &_contract);
+	solAssert(it != hierarchy.end(), "Base not found in inheritance hierarchy.");
+	++it;
+	if (it == hierarchy.end())
+		return nullptr;
+	else
+	{
+		solAssert(*it != &_contract, "");
+		return *it;
+	}
+}
+
+FunctionDefinition const* IRGenerationContext::nextConstructor(ContractDefinition const& _contract) const
```

Comment:
This can be part of ContractDefinition with passing mostDerivedContract as parameter